packages/kolme-cli/src/chain_query.rs

```diff
@@ -0,0 +1,223 @@
+use anyhow::{anyhow, bail, ensure, Context, Result};
+use reqwest::{Client, Url};
+use serde::Deserialize;
+
+use crate::RequestBuilderExt;
+
+pub struct ChainApi {
+    client: Client,
+    url: Url,
+}
+
+#[derive(Deserialize)]
+#[allow(dead_code)]
+pub struct ChainVersion {
+    pub code_version: String,
+    pub chain_version: String,
+    pub next_height: BlockHeight,
+}
+
+#[allow(dead_code)]
+pub struct BlockResponse {
+    pub code_version: String,
+    pub chain_version: String,
+    pub block_height: BlockHeight,
+}
+
+pub struct ForkInfo {
+    pub first_block: BlockHeight,
+    pub last_block: BlockHeight,
+}
+
+#[derive(Deserialize, Copy, Clone, Debug)]
+pub struct BlockHeight(pub u32);
+
+impl BlockHeight {
+    pub fn pred(&self) -> Result<BlockHeight> {
+        let pred = self.0.checked_sub(1).context("Underflow in pred")?;
+        Ok(BlockHeight(pred))
+    }
+
+    pub fn succ(&self) -> Result<BlockHeight> {
+        let succ = self.0.checked_add(1).context("Overflow in succ")?;
+        Ok(BlockHeight(succ))
+    }
+
+    pub fn increasing_middle(&self, block_height: BlockHeight) -> Result<BlockHeight> {
+        ensure!(
+            self.0 <= block_height.0,
+            "Cannot compute middle since start_block is greater than end_block"
+        );
+
+        let middle = block_height
```

Comment:
NIT: middle seems like a confusing name here, since it seems like the delta to the middle, not the midpoint itself.